### DISCUSSION TITLE
How to disable pyright analysis for some dependencies?

### DISCUSSION THREAD

**Discussion by tamuhey** [microsoft/pyright#2124] [Q&A] [answered]
Analyzing [Transformers](https://github.com/huggingface/transformers) seems to take a very long time, and it significantly slows down the autocompletion.
Thus, I want to disable Pyright analysis for Transformers. What should I do?

**Comment by erictraut** [comment, marked as answer]
I assume that you have `useLibraryCodeForTypes` enabled? If so, you could disable it. By default, it's off in pyright (but on in pylance). That would explain why you're seeing poor performance because this is an untyped library, so pyright needs to do significant work to try to infer types from the source.

Another option is to generate type stubs for this library. You can either use pyright's built-in stub generation or another tool like stubgen. If you have stubs (even ones that are incomplete), pyright will use them rather than attempting to extract type information from the source code.

**Reply by tamuhey** [reply]
Thanks!
